app/models/user-prefs.js

```diff
@@ -0,0 +1,56 @@
+import Ajv from 'ajv';
+import { cloneDeep } from 'lodash';
+
+import { addModel as commentModelMaker } from './comment';
+
+const commentModel = commentModelMaker(null);
+
+const schema = {
+  '$schema': 'http://json-schema.org/schema#',
+  id:        'https://freefeed.net/schemas/user-preferences-schema',
+
+  type:       'object',
+  properties: {
+    hideCommentsOfTypes: {
+      title:   `Do not show comments with these hideType's`,
+      default: [],
+
+      type:        'array',
+      uniqueItems: true,
+      items:       {
+        type: 'integer',
+        enum: [
+          commentModel.DELETED,
+          commentModel.HIDDEN_BANNED,
+          commentModel.HIDDEN_ARCHIVED,
+        ],
+      },
+    },
+  },
+  additionalProperties: false,
+};
+
+const ajv = new Ajv({ allErrors: true, useDefaults: true });
+const check = ajv.compile(schema);
+
+/**
+ * Validates and completes absent fields by default values.
+ * Throws exception if validation failed (or returns default
+ * value if 'safe' argument is true).
+ * 
+ * @param {object}  data
+ * @param {boolean} safe - return default value instead of exception throwing
+ * @return {object}
+ */
+export function valiate(data = {}, safe = false) {
+  data = cloneDeep(data);
+  const valid = check(data);
+  if (valid) {
+    return data;
+  }
```

Comment:
Is there any reasone not to use `ajv` in async mode? As far as I can see from the docs, it is the recommended way of dealing with it